packages/substyle/src/createUseStyle.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import { useContext, useMemo } from 'react'
+import { type KeysT, type PropsT } from './types'
+import createSubstyle from './createSubstyle'
+import coerceSelection from './coerceSelection'
+import { PropsDecoratorContext } from './PropsDecoratorProvider'
+
+type DependsOnFuncT = (props: Object) => any[]
+
+const createUseStyle = (
+  defaultStyle?: Object | ((props: Object) => Object),
+  getModifiers?: (props: Object) => KeysT,
+  getDependsOn: DependsOnFuncT = props => Object.values(props)
+) => {
+  const useStyle = (props: PropsT) => {
+    const { style, className, classNames, ...rest } = props
+    const propsDecorator = useContext(PropsDecoratorContext)
+
+    const substyle = useMemo(
+      () => createSubstyle({ style, className, classNames }, propsDecorator),
+      [style, className, classNames, propsDecorator]
+    )
+
+    const modifiers = getModifiers ? coerceSelection(getModifiers(rest)) : []
+    return useMemo(
+      () =>
+        substyle(
+          modifiers,
+          typeof defaultStyle === 'function' ? defaultStyle(rest) : defaultStyle
+        ),
+      [
+        substyle,
+        ...modifiers,
```

Comment:
This dynamic result of active modifiers leads to a React warning:

```
Warning: The final argument passed to useMemo changed size between renders. The order and size of this array must remain constant.
```

Reply:
possible solution: sort to keep order stable, then concatenate

Reply:
probably sorting is too slow to be useful as perf optimization. modifiers reordering is also rather rare, I'd guess. so simply concatenating should be good enough